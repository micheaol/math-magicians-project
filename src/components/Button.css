button {
    border: 0.52px solid white;
    background-color: rgb(222, 220, 233);
    font-size: 24px;
    color: rgb(36, 34, 34);
    font-weight: bold;
    cursor: pointer;
    outline: none;
    padding-top: 9%;
  }

  button:nth-child(4n){
    background-color: rgb(244,152,60);
  }

  button:nth-last-child(1){
    background-color: rgb(244,152,60);
  }

  button:nth-last-child(3){
    grid-column: 1 / span 2;
    grid-row: 5;
  }
  
  button:hover {
    background-color: rgb(92, 92, 97);
  }
