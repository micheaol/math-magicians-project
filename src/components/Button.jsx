/* eslint-disable linebreak-style */
/* eslint-disable arrow-body-style */
/* eslint-disable react/prop-types */

import React from 'react';
import './Button.css';

// eslint-disable-next-line react/prefer-stateless-function
const Button = ({ className, value, onClick }) => {
  return (
    <button className={className} onClick={onClick} type="submit">
      {value}
    </button>
  );
};

export default Button;
