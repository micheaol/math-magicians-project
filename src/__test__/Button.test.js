import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

afterEach(() => {
  cleanup();
});
describe('render button component to the screen', () => {
  test('should render button component', () => {
    render(<Button />);
    const buttonElement = screen.getByTestId('button');
    expect(buttonElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
