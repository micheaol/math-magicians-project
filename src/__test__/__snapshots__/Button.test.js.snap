// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button component to the screen matches snapshot 1`] = `
<button
  data-testid="button"
  type="submit"
/>
`;
